src/types/generic.ts

```diff
@@ -0,0 +1,262 @@
+   *                                    This property is added by the custom strategy.
+   * @property {string} [type=default] The type of area card.
+   */
+  export interface areaEntity {
+    aliases?: string[];
+    area_id: string | null;
+    name: string;
+    picture: string | null;
+    order?: number;
+    hidden?: boolean;
+    extra_cards?: object[];
+    type?: string;
+  }
+
+  /**
+   * View Entity.
+   *
+   * This entity is added by the custom strategy.
+   *
+   * @property {string} title Title of the entity.
+   * @property {string} icon Icon to use for the entity in the frontend.
+   *                         Example: `mdi:home`.
+   * @property {number} [order] Ordering position of the entity in the list of available views.
+   * @property {boolean} [hidden] True if the entity should be hidden from the dashboard.
+   */
+  export interface viewEntity {
+    hidden?: boolean;
+    icon: string;
+    order?: number;
+    title: string;
+  }
+
+  /**
+   * Domain Entity.
+   *
+   * This entity is added by the custom strategy.
+   *
+   * @property {number} [order] Ordering position of the entity in the list of available views.
+   * @property {boolean} [hidden] True if the entity should be hidden from the dashboard.
+   */
+  export interface domainEntity extends titleCardOptions {
+    hidden?: boolean;
+    order?: number;
+  }
+
+  /**
+   * Title Card options.
+   *
+   * @property {string} [title] Title to render, which may contain templates.
+   * @property {string} [subtitle] Subtitle to render, which may contain templates.
+   * @property {boolean} [showControls=true] False to hide controls.
+   * @property {string} [iconOn] Icon to show for switching entities from off state.
+   * @property {string} [iconOff] Icon to show for switching entities to off state.
+   * @property {string} [onService=none] Service to call for switching entities from off state.
+   * @property {string} [offService=none] Service to call for switching entities to off state.
+   */
+  export interface titleCardOptions {
+    iconOff?: string;
+    iconOn?: string;
+    offService?: string;
+    onService?: string;
+    showControls?: boolean;
+    subtitle?: string;
+    title?: string;
+  }
+
+  /**
+   * Dashboard Information Object.
+   *
+   * @property {LovelaceConfig} config User supplied dashboard configuration, if any.
+   * @property {HomeAssistant} hass The Home Assistant object.
+   * @property {boolean} narrow If the current user interface is rendered in narrow mode or not.
+   *
+   * @see https://developers.home-assistant.io/docs/frontend/custom-ui/custom-strategy/#dashboard-strategies
+   */
+  export interface dashBoardInfo {
+    config?: LovelaceConfig;
+    hass: HomeAssistant;
+    narrow: boolean;
+  }
+
+  export interface LovelaceConfig {
+    title?: string;
+    strategy?: {
+      type: string;
+      [key: string]: any;
+    };
+    views: LovelaceViewConfig[];
+    background?: string;
+  }
```

Comment:
This can be removed maybe? Since it can be imported from custom-card-helpers

Reply:
ah, I see you added strategy. You should probably extend LovelaceConfig instead. E.g.
```
export interface Config extends LovelaceConfig {
    strategy?: {
      type: string;
      [key: string]: any;
    };
  }
```

Reply:
custom-card-helpers haven't been updated for two years now.
I'm working on removing this dependency.